bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -368,6 +369,7 @@ void PreProcessor::cancelPreProcessing(NodeIdType clientId) {
       reqSeqNum = clientEntry->reqProcessingStatePtr->getReqSeqNum();
       releaseClientPreProcessRequest(clientEntry, clientId, CANCEL);
       LOG_WARN(logger(), "Pre-processing consensus not reached - abort request " << KVLOG(reqSeqNum, clientId));
+      preProcessorMetrics_.currentPreProcReqNum.Get().Dec();
```

Comment:
Please increment this metric in registerRequest and decrement in releaseClientPreProcessRequest. No other places should be changed :-)